### ISSUE TITLE
Incompatible with GraphQL v15 & Typescript

### ISSUE THREAD

**Issue by Lexicality** [Urigo/graphql-scalars#2105]
### Issue workflow progress

_Progress of the issue based on the [Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [ ] 1. The issue provides a reproduction available on GitHub, Stackblitz or CodeSandbox
> Make sure to fork this template and run `yarn generate` in the terminal.
>
> Please make sure the GraphQL Scalars package version under `package.json` matches yours.
- [ ] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [ ] 4. A pull request is pending review

---

**Describe the bug**

Same as #1522 and #1195
```
node_modules/graphql-scalars/typings/scalars/iso-date/Time.d.ts:10:35 - error TS2315: Type 'GraphQLScalarType' is not generic.

10 export declare const GraphQLTime: GraphQLScalarType<Date, string>;
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**To Reproduce**
- Install into a project using GraphQL 15
- Run `npx tsc`

**Expected behavior**

Typescript successfully builds

**Environment:**

- OS: Ubuntu
- GraphQL Scalars Version: v1.22.2
- GraphQL Version: v15.8.0
- Typescript version: v5.1.6
- NodeJS: v18.17.0

**Additional context**

**Comment by acomito** [comment]
@Lexicality did you ever fix this?

What combo worked for you?

**Comment by Lexicality** [comment]
> @Lexicality did you ever fix this?
> 
> What combo worked for you?

Nope! I simply gave up and used a variety of other libraries and my own code

**Comment by mmmulani** [comment]
seems like you need graphQL version 16: https://github.com/graphql/graphql-js/commit/4f21cdc1f1cbd5873735b8426f8aa8b2f8d5db8c

breaking change from graphql-scalars that takes advantage of it: https://github.com/Urigo/graphql-scalars/commit/0fb119aa21ef552ed00a65ea7ab813ff06473cc6

downgrading to graphql-scalars@1.21.3 fixes this

**Comment by Lexicality** [comment]
In that case it would be nice if the peer dependencies were updated to make that clearer.
